shtdlib.sh

```diff
@@ -1996,13 +1996,44 @@ function ln_sf {
 # Create string of random characters
 #  - First param is length, default: 20
 #  - Second param is characters, default: A-Za-z0-9_ (Note: '-' specifies range)
-gen_rand_chars() {
+function gen_rand_chars {
+    shopt_decorator_option_name='pipefail'
+    shopt_decorator_option_value='false'
+    # shellcheck disable=2015
+    shopt_decorator "${FUNCNAME[0]}" "${@:-}" && return || conditional_exit_on_fail 121 "Failed to run ${FUNCNAME[0]} with shopt_decorator"
+
     local length="${1:-20}"
     local chars="${2:-A-Za-z0-9_}"
     debug 10 "Creating a string of random characters of length: ${length} and chars: ${chars}"
     LC_CTYPE=C tr -dc "${chars}" < '/dev/urandom' | head -c "${length}"
 }
 
+# Checks if an environtment variable is set and contains a string longer than
+# 0, if not then it's set to a random value.
+# If a file name/path is specified then a line containing VARIABLE=VALUE is
+# written to the end of the file. Optionally the length of the random
+# string/value can be specified. (defaults to 50)
+function auto_ensure_key_exists {
+    local file_path="${1}"
+    local var_name="${2}"
+    local key_length="${3:-50}"
```

Comment:
Should the key length value be configurable? 50 seems like a large value as a default length.

Reply:
key length is configurable yes, 50 is just what we've been using, since it's never entered manually I think it's fine unless there is a particular reason to change it?

Reply:
Default in gen_rand_chars is 20. I'm not against a 50 character password I just wouldn't want to have to manually type it out if copy/paste isn't available.